packages/core/src/http-client/http-client.ts

```diff
@@ -139,14 +147,21 @@ export async function buildAxiosRequestConfig<T extends HttpRequestConfig>(
  * @param destination - A destination or a destination name and a JWT.
  * @param requestConfig - Any object representing an HTTP request.
  * @param options - An [[HttpRequestOptions]] of the http request for configuring e.g., csrf token delegation. By default, the SDK will not fetch the csrf token.
+ * @param destinationOptions - Options to employ when fetching destinations.
  * @returns A promise resolving to an [[HttpResponse]].
  */
 export function executeHttpRequest<T extends HttpRequestConfig>(
   destination: Destination | DestinationNameAndJwt,
   requestConfig: T,
-  options?: HttpRequestOptions
+  options?: HttpRequestOptions,
+  destinationOptions?: DestinationOptions
```

Comment:
Two optional parameters can be inconvenient, if you need the second one.
We actually already talked about this in the past and decided not to add this.
We might want to consider writing it down somewhere (ADR?)
Please check this [comment](https://github.com/SAP/cloud-sdk-js/issues/1262#issuecomment-848918155).

The same of course applies to the other functions here.

Reply:
I also had the feeling that this is not a nice API. But I was not aware we made it a clear `no go`, I adjusted the API.